### PR TITLE
feat: enable openapi spec from url in api register

### PR DESCRIPTION
**Pull request by frankqianms**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [ ] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/apic-extension/azext_apic_extension/_help.py [hunk 1]
```diff
@@ -31,4 +31,7 @@
         text: |
           az apic api register -g api-center-test -n contosoeuap --api-location "examples/cli-examples/spec-examples/openai.json" --environment-id public
           az apic api register -g api-center-test -n contosoeuap --api-location "examples/cli-examples/spec-examples/openai.yml" --environment-id public
+      - name: Register api by providing spec url.
+        text: |
+          az apic api register -g api-center-test -n contosoeuap --api-location "https://petstore.swagger.io/v2/swagger.json" --environment-id public
```
Comment (frankqianms) [inline src/apic-extension/azext_apic_extension/_help.py:36]: Do we need to add `az apic api register -g api-center-test -n contosoeuap --api-location "https://petstore.swagger.io/v2/swagger.yaml" --environment-id public` to show that both yaml and json format from url can be supported

File: src/apic-extension/azext_apic_extension/tests/latest/test_register_command.py [hunk 2]
```diff
@@ -116,6 +116,78 @@ def test_register_with_json_spec(self):
         finally:
             os.remove(exported_file_path)
 
+    @ResourceGroupPreparer(name_prefix="clirg", location=TEST_REGION, random_name_length=32)
+    @ApicServicePreparer()
+    def test_register_with_json_spec_from_url(self):
```
Comment (blackchoey) [inline src/apic-extension/azext_apic_extension/tests/latest/test_register_command.py:121]: please also add some test cases to test the error handling logic (e.g. when url is invalid)

File: src/apic-extension/azext_apic_extension/custom.py [hunk 3]
```diff
@@ -162,26 +162,49 @@ def register_apic(cmd, api_location, resource_group, service_name, environment_i
...
+        if str(api_location).startswith('https://') or str(api_location).startswith('http://'):
+            try:
+                # Fetch the content from the URL
+                response = requests.get(api_location)
+                # Raise an error for bad status codes
+                response.raise_for_status()
+                # Try to parse the content as JSON
+                try:
+                    data = json.loads(response.content)
+                except json.JSONDecodeError:
```
[the remark this answers is not in the document]
Comment (blackchoey) [reply src/apic-extension/azext_apic_extension/custom.py:177]: What do you mean by "validate the link"?

[the remark this answers is not in the document]
Comment (frankqianms) [reply src/apic-extension/azext_apic_extension/custom.py:177]: do we need to load the content and to validate the content

File: src/apic-extension/azext_apic_extension/tests/latest/test_register_command.py [hunk 4]
```diff
@@ -188,6 +189,42 @@ def test_register_with_yaml_spec_from_url(self):
...
+    @ApicServicePreparer()
+    def test_register_using_spec_url_with_invalid_content(self):
+        # Set up an URL with invalid spec
+        self.kwargs.update({
+            'spec_url': 'https://github.com/'
+        })
+        # Capture the logs
+        with self.assertLogs(logger, level='ERROR') as log:
+            with self.assertRaises(SystemExit) as cm:
+                self.cmd('az apic api register -g {rg} -n {s} -l "{spec_url}"', expect_failure=True)
```
Comment (blackchoey) [inline src/apic-extension/azext_apic_extension/tests/latest/test_register_command.py:220]: Will this line throw an exception even we set `expect_failure=True`?

**Comment by frankqianms** [comment]
Error messages will be like
image

**Review by blackchoey — APPROVED** [review]

**Pull request merged**